viewing_party/party.py

```diff
@@ -1,23 +1,137 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    new_movie = {}
+
+    if title and genre and rating:
+        new_movie["title"] = title
+        new_movie["genre"] = genre
+        new_movie["rating"] = rating
+        return new_movie
+    return None
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for movie in user_data["watchlist"]:
+        if title == movie["title"]:
+            add_to_watched(user_data, movie)
+            user_data["watchlist"].remove(movie)
+    return user_data
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    avg_rating = 0.0
+    summed_ratings = 0
+    collected_ratings = []
+
+    if len(user_data["watched"]) >= 1:
+        collected_ratings = [movie["rating"] for movie in user_data["watched"]]
+    else:
+        return avg_rating
+
+    for rating in collected_ratings:
+        summed_ratings += rating
+
+    avg_rating = (summed_ratings / len(collected_ratings))
+    return avg_rating

```

Comment:
I know we hadn't covered time and space complexity when you wrote this, but while this solution works, it's a little less efficient in space than you can be. Since `collected_ratings` will eventually have as many entries as `user_data["watched"]`, this is O(n) in space complexity.

By calculating the sum yourself, you can move it down to O(1):

```python
summed_ratings = 0.0

for movie in user_data["watched"]:
    summed_ratings += movie["rating"]

avg_rating = summed_ratings / len(user_data["watched"])
return avg_rating
```

Not a major problem, of course, especially since it was very unlikely to be on your mind at the time!

Note that this still manages to return `0.0` when `user_data["watched"]`  is falsy as the `for` loop will not be entered so `summed_ratings` will be `0.0`, which means `avg_rating` will be too.

(Also note that true to my advice above about when to define variables, I don't define `avg_rating` until I need it. You can also see that the parentheses around the division is not needed, but that's even more minor.)